### PR TITLE
feat: shared workflow config lib + hook config-awareness

### LINKED ISSUE #779: feat: shared workflow config lib + hook config-awareness
## Summary

Create a shared `lib/workflow-config.js` module and update existing hooks to respect `workflow.*` toggles. When a workflow step is disabled in config, its corresponding hooks become no-ops.

## Shared module: `lib/workflow-config.js`

Reads `.dev-team/config.json` once and exposes workflow state:

```js
const { isEnabled } = require("./lib/workflow-config");
if (!isEnabled("review")) process.exit(0); // skip when review disabled
```

Handles missing config, missing workflow section, and individual missing keys gracefully (defaults to enabled).

## Hooks that gain config-awareness

| Hook | Config key | Behavior when `false` |
|------|-----------|----------------------|
| `post-change-review.js` | `workflow.review` | Don't fire — don't spawn review agents |
| `review-gate.js` | `workflow.review` | Don't gate — allow commits without review evidence |
| `merge-gate.js` | `workflow.review` | Skip review-evidence check (still validate other merge prerequisites) |
| `pre-commit-gate.js` | `workflow.learn` | Skip memory freshness check — allow commits without memory updates |

## What stays unchanged

These hooks are universal and workflow-independent:
- `safety-guard.js` — always active
- `pre-commit-lint.js` — always active
- `tdd-enforce.js` — always active
- `worktree-create.js` / `worktree-remove.js` — infrastructure workaround
- `agent-teams-guide.js` — advisory
- `watch-list.js` — user-configured patterns

## Design: Option A (shared config guard)

Each existing hook adds one line at the top:
```js
const { isEnabled } = require("./lib/workflow-config");
if (!isEnabled("review")) process.exit(0);
```

Hooks stay as separate files with single responsibility. The shared module DRYs the config reading.

## Related

Depends on: #773 (workflow config schema)

---

### Hook language-neutrality (from code audit)

`pre-commit-lint.js` and `tdd-enforce.js` have JS/TS-specific patterns that should be generalized:

**`pre-commit-lint.js`:**
- Primarily checks `package.json` for lint/format scripts via `npm`
- Has a Python fallback (ruff via `pyproject.toml`) but nothing for Go (`golangci-lint`), Rust (`cargo clippy`/`cargo fmt`), Ruby (`rubocop`), etc.
- The config lib from this issue should expose ecosystem detection that the lint hook can use

**`tdd-enforce.js`:**
- `node_modules` skip is hardcoded — should generalize to a set of vendor/build directories per ecosystem (`vendor/`, `target/`, `.venv/`, `__pycache__/`, etc.)

### Shared types for hooks (from #782 research)

The `lib/workflow-config.js` shared module should also export types for sidecar structures (`SidecarFile`), workflow config (`WorkflowConfig`), and PR config (`PrConfig`) that hooks consume. This enables type-safe config reading in TS hooks (after #782 migration).

### Related research

- `docs/research/782-ts-hook-compilation-2026-04-04.md`

### PR DESCRIPTION
**Pull request by fredericboyer**
## Summary
- Created `templates/hooks/lib/workflow-config.js` — shared module that reads `workflow.*` toggles from `.dev-team/config.json` with safe defaults (missing config = enforce)
- Updated 4 hooks to respect config toggles: `post-change-review` (`workflow.review`), `review-gate` (`workflow.review`), `merge-gate` (`workflow.review`), `pre-commit-gate` (`workflow.learn`)
- Added 11 unit tests covering all edge cases: missing config, malformed JSON, missing keys, symlink rejection, caching

Closes #779

## Test plan
- [x] All 834 existing tests pass
- [x] 11 new workflow-config tests pass
- [x] Verified safe defaults: missing config/section/key all default to enabled (enforce)
- [x] Only explicit `false` disables a workflow feature
- [x] Symlinked config files are rejected (defaults to enforce)

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/unit/workflow-config.test.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
+"use strict";
+
+const { describe, it, beforeEach, afterEach } = require("node:test");
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply tests/unit/workflow-config.test.js]: Fixed: removed unused beforeEach/afterEach imports.
